ServiceApp/src/org/droidplanner/services/android/core/gcs/follow/Follow.java

```diff
@@ -33,22 +39,28 @@
 
     private final LocationFinder locationFinder;
     private FollowAlgorithm followAlgorithm;
+    private final LocationRelay mLocationRelay;
 
-    public Follow(MavLinkDroneManager droneMgr, Handler handler, LocationFinder locationFinder) {
+    public Follow(Context context, MavLinkDroneManager droneMgr, Handler handler, LocationFinder locationFinder) {
         this.droneMgr = droneMgr;
         final MavLinkDrone drone = droneMgr.getDrone();
         if(drone != null)
             drone.addDroneListener(this);
 
-        followAlgorithm = FollowAlgorithm.FollowModes.LEASH.getAlgorithmType(droneMgr, handler);
+        followAlgorithm = (drone instanceof ArduSolo)?
+                FollowAlgorithm.FollowModes.SOLO_SHOT.getAlgorithmType(droneMgr, handler):
+                FollowAlgorithm.FollowModes.LEASH.getAlgorithmType(droneMgr, handler);
 
         this.locationFinder = locationFinder;
         locationFinder.addLocationListener(TAG, this);
+
+        mLocationRelay = new LocationRelay();
     }
 
-    public void toggleFollowMeState() {
+    public void toggleFollowMeState(FollowLocationSource source) {
```

Comment:
@kellyschrock the method parameter `source` is not being used in the method body.